titus-server-master/src/main/java/com/netflix/titus/master/kubernetes/controller/NodeGcController.java

```diff
@@ -148,9 +147,15 @@ private boolean isAgentInstanceNotAvailable(String nodeName) {
 
         Optional<AgentInstance> agentInstanceOpt = agentManagementService.findAgentInstance(nodeName);
         if (!agentInstanceOpt.isPresent()) {
-            return true;
+            // re-check with agent service to find an instance
+            AgentInstance agentInstance = agentManagementService.getAgentInstanceAsync(nodeName)
+                    .toBlocking()
+                    .firstOrDefault(null);
+            if (agentInstance != null) {
+                return agentInstance.getLifecycleStatus().getState() == InstanceLifecycleState.Stopped;
```

Comment:
Can LifecycleStatus be `null`?

Reply:
Not sure, but an additional check won't hurt